src/ripple/app/paths/impl/XRPEndpointStep.cpp

```diff
@@ -360,6 +361,18 @@ XRPEndpointStep<TDerived>::check (StrandContext const& ctx) const
     if (ter != tesSUCCESS)
         return ter;
 
+    if (ctx.view.rules().enabled(fix1781))
+    {
+        auto const issuesIndex = isLast_ ? 0 : 1;
+        if (!ctx.seenDirectIssues[issuesIndex].insert(xrpIssue()).second)
+        {
+            JLOG(j_.debug())
+                << "XRPEndpointStep: loop detected: Index: " << ctx.strandSize
+                << ' ' << *this;
+            return temBAD_PATH_LOOP;
```

Comment:
When I run code_cov over the new code, this line returning `temBAD_PATH_LOOP` isn't hit.  Is there a (not too painful) way to hit this line with a unit test?

Reply:
I don't think I can make a test that will hit this. The reason is the book steps also check for a path, and the book step will detect the loop before the endpoint will.

I'd still like to leave this code in, but in practice I don't think it can be hit.